### DISCUSSION TITLE
Feedback value for Auto Throttle toggle button

### DISCUSSION THREAD

**Discussion by Jaro54** [nguyenquyhy/Flight-Tracker-StreamDeck#411] [General]
Hello, I am new to Flight Tracker StreamDeck. I created many buttons in Stream Deck XL (see picture) without difficulties. However I have problems to get the right feedback for the Auto Throttle for the Cessna Citation Longitude plane. By using toggle event: AUTO_THROTTLE_ARM the Auto Throttle function will go on/of. So that works. But I am not able to find the correct feedback value. By using AUTO THROTTLE ARM I get feedback for the Off and On position. Same issue if I use On and Off images. 
![Screenshot 2023-11-26 142037](https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/assets/152080175/98ffc563-c7a3-4adf-b943-88ef24a9f2bb)

**Comment by nguyenquyhy** [comment]
I'm a bit confused here. Are you trying to get a 3 state feedback ON/OFF/ARM?

**Reply by Jaro54** [reply]
No, just ON/OFF. Or armed/disarmed.

**Reply by Jaro54** [reply]
More info: I get the feedback On image in the Off state only. Every other image I used for the On state is not displayed at the button, but the image of the Off state is still displayed. In msfs2020 A/T function works when I toggle On/Off.

**Reply by nguyenquyhy** [reply]
From SimConnect documentation, `AUTOPILOT THROTTLE ARM` is the default display variable to be used. However, the Longitude implements custom Autothrottle logic and uses its own LVars.

You can get proper ON/OFF image by putting this to feedback value.
```
L:WT_Longitude_Autothrottle_Status==2
```

Please note that LVar might be changed by WorkingTitle any time in the future.

**Reply by Jaro54** [reply]
Thanks, I will try tomorrow. Now it is time to go to bed.

**Reply by Jaro54** [reply]
Sorry, it is not working. I also tried without underscores, because it is a variable, I think.
![Screenshot 2023-11-27 103954](https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/assets/152080175/6c2fe66d-dec7-498c-8b52-46af453700b5)

**Reply by nguyenquyhy** [reply]
Are you sure you are using the latest version of the plugin? Maybe you can export the profile and share here too.

**Reply by Jaro54** [reply]
I cannot see the version number, but I downloaded the plugin on November 12, so that should be 0.51. Where can I find the profile on my Win PC?

**Reply by nguyenquyhy** [reply]
https://docs.elgato.com/en/stream-deck/profiles/exporting

**Reply by Jaro54** [reply]
That was easy. I cannot attach this file type, so I created a ZIP file.
[AUTO PILOT.zip](https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/files/13473102/AUTO.PILOT.zip)

**Reply by nguyenquyhy** [reply]
Hmm that's very strange. I use that exact same setup and it works for me. Are you using the default Longitude or some other mods?

A few things to try
1. Try `L:WT_Longitude_Autothrottle_Status==3` instead
2. Or put `L:WT_Longitude_Autothrottle_Status` in Display Value and see what is the raw value when you arm AT

**Reply by Jaro54** [reply]
Sorry, I had lunch first. Yes, both work. Thank you very much for your time and effort. I have a question: is it possible I can view the code of the scripts so I can learn from it. Is it written in C++?

**Reply by nguyenquyhy** [reply]
All the code is in this repository. Complete open source :).
